builder/parser/utils.go

```diff
@@ -50,6 +50,8 @@ func fullDispatch(cmd, args string) (*Node, map[string]bool, error) {
 // splitCommand takes a single line of text and parses out the cmd and args,
 // which are used for dispatching to more exact parsing functions.
 func splitCommand(line string) (string, string, error) {
+	// Make sure we get the same results irrespective of leading/trailing spaces
+	line = strings.TrimSpace(line)
 	cmdline := TOKEN_WHITESPACE.Split(line, 2)
```

Comment:
the latter

> On Jan 15, 2015, at 12:25 PM, Doug Davis <notifications@github.com> wrote:
> 
> In builder/parser/utils.go <https://github.com/docker/docker/pull/10120#discussion-diff-23040804>:
> 
> > @@ -50,6 +50,8 @@ func fullDispatch(cmd, args string) (*Node, map[string]bool, error) {
> >  // splitCommand takes a single line of text and parses out the cmd and args,
> >  // which are used for dispatching to more exact parsing functions.
> >  func splitCommand(line string) (string, string, error) {
> > +	// Make sure we get the same results irrespective of leading/trailing spaces
> > +	line = strings.TrimSpace(line)
> >  	cmdline := TOKEN_WHITESPACE.Split(line, 2)
> Can you elaborate? Do you mean that you want me to modify TOKEN_WHITESPACE so that the call to Split() will remove the need for the TrimSpaces() call? If so, I'm not sure how to do that :-)
> 
> Or did you just mean:
> cmdLine := TOKEN_WHITESPACE.Split(strings.TrimSpace(line), 2)
> which I can do if you want.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/10120/files#r23040804>.
>